src/utils/structures/ClientUtils.ts

```diff
@@ -46,6 +46,11 @@ export class ClientUtils {
     }
 
     public async getChannelCount(textOnly = false, voiceOnly = false): Promise<number> {
+        /**
+         * This method is throwing reference error
+         * ChannelType is not defined, even though it is being imported
+         * for now i added channel types directly instead of using enums
+         */
```

Comment:
This needs to be solved, ofc

Reply:
this error is so confusing. I imported ChannelType from discord.js correctly but still it throws reference error
and because of this error the setPresence method is not working.
what i did now is instead of using `ChannelType.GuildText` i used `0` and it's working.

Reply:
Finally found the cause of the error. Apparently, I just remembered that `broadcastEval` stringifies the function to be evaluated by shards. But since there's no `ChannelType` variable in the environment where the function is being evaluated, the code emits an error saying that `ChannelType` is not defined.

We can fix this by adding `ChannelType` to the context (see second argument of `broadcastEval`):
```javascript
{ context: { textOnly, voiceOnly, types: ChannelType } }
```

In the function, there's already `t` variable as the second parameter. That's the context. So, in the function, to access the channel types you can access `types` property from the context:
```javascript
if (t.textOnly) {
    return ch.type === t.types.GuildText ||
        ch.type === t.types.PublicThread ||
        ch.type === t.types.PrivateThread;
} else if (t.voiceOnly) {
    return ch.type === t.types.GuildVoice;
}
```

Reply:
wow, i was working on same file and found same fix :joy:

Reply:
wait, really? wow 🤣